NYPL/MyLibraryNYCApp
# Tgr 141 set up db automatically

## JIRA tickets:
- [My Library NYC: Set up database automatically](https://newyorkpubliclibrary.atlassian.net/browse/TGR-141)

## Things Done:
- Added a task to set up the test and development databases locally when they haven't been set up already
- Mounted the postgres database in `data/` for easier access and removal

## How to Test:
- In your existing local application, bring up a webapp task with `docker-compose run webapp bash`. Do `RAILS_ENV=test bundle exec rake db:drop` and then also `RAILS_ENV=development bundle exec rake db:drop`
- Exit the container and spin it down.
- Rebuild the app with `docker-compose build`
- Bring the app up with `docker-compose run`. This should set up the databases.
- The app should serve successfully at localhost:3000 with the default seeded data
- Bring down the app and bring it back up. This time, there should not be a bunch of log messages related to loading the database schema and seeding data. The app should still work as normal.
- Do `RAILS_ENV=test bundle exec rails test`. All the tests should pass.
